From: oulijun <oulijun@huawei.com>
Date: Fri, 4 May 2018 10:57:07 +0800
Subject: RDMA/hns: Add return operation when configured global param fail
Patch-mainline: v4.17-rc7
Git-commit: 2349fdd483ea933b223f3eca53ed42835383d316
References: bsc#1104427 FATE#326416

When configure global param function run fail, it should directly return
and the initial flow will stop.

Signed-off-by: Lijun Ou <oulijun@huawei.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/hns/hns_roce_hw_v2.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
@@ -1085,6 +1085,7 @@ static int hns_roce_v2_profile(struct hn
 	if (ret) {
 		dev_err(hr_dev->dev, "Configure global param fail, ret = %d.\n",
 			ret);
+		return ret;
 	}
 
 	/* Get pf resource owned by every pf */
